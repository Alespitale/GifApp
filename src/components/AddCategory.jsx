import { useState } from "react";

export const AddCategory = ({ addCategory, categories, danger }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = ({ target }) => setInputValue(target.value);

  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length < 1) return;
    if (categories.includes(inputValue)) return alert("You already searched that!");
    addCategory(inputValue.trim());
    setInputValue("");
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        type="text"
        placeholder="Search GIF"
        value={inputValue}
        onChange={handleInputChange}
      />
    </form>
  );
};
