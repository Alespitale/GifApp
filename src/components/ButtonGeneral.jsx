import styled from '@emotion/styled';

export const ButtonGeneral = styled.button`
  padding: 12.5px;
  background-color: black;
  font-size: 15px;
  border-radius: 3rem;
  color: white;
  border: 1px solid rgb(97, 32, 158);
  &:hover {
    color: black;
    background: pink;
  }
`;
