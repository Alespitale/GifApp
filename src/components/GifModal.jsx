import Modal from "react-bootstrap/Modal";
import { getDownload } from "../helpers/getDownload";
import PropTypes from "prop-types";
import { DownloadIcon } from "./icons/Download.jsx";
import { CloseIcon } from "./icons/Close.jsx";

export const GifModal = ({ url, title, show, onHide }) => {
  return (
      <Modal className="modal-content" show={show}>
        <Modal.Header className="modal-header">
          <Modal.Title className="modal-title">{title}</Modal.Title>
          <button
            data-testid="close button"
            className="close-btn"
            onClick={onHide}
          >
            <CloseIcon />
          </button>
        </Modal.Header>
        <Modal.Body>
          <div>
            <img src={url} alt={title} />
          </div>
          <button
            data-testid="download button"
            className="download-btn"
            onClick={() => getDownload({ url, title })}
          >
            <DownloadIcon />
          </button>
        </Modal.Body>
      </Modal>
  );
};

GifModal.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  show: PropTypes.bool.isRequired,
  onHide: PropTypes.func.isRequired,
};
