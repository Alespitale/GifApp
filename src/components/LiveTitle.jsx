import "../styles/LiveTitle.css";

export const LiveTitle = () => {
  return (
    <div className="title-container">
      <svg viewBox="0 0 960 100">
        <symbol id="s-text">
          <text textAnchor="middle" x="50%" y="80%">
            Gifinder
          </text>
        </symbol>
        <g className="g-ants">
          <use xlinkHref="#s-text" className="title"></use>
          <use xlinkHref="#s-text" className="title"></use>
          <use xlinkHref="#s-text" className="title"></use>
          <use xlinkHref="#s-text" className="title"></use>
          <use xlinkHref="#s-text" className="title"></use>
        </g>
      </svg>
    </div>
  );
};
