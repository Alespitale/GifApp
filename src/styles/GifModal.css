.modal-content {
  align-items: center;
  position: fixed;
  background-color: rgba(0, 0, 0, 0.50);
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
}

.modal-content img {
  width: 100%;
  padding: 10px;
  border-radius: 25px;
  box-shadow: 0px 3px 5px rgba(0, 0, 0, 0.3);
}

.modal-title {
  font-size: 1rem;
  max-width: 75%;
  text-align: center;
  color: white;
  text-transform: uppercase;
}

.modal-header {
  width: 100%;
  max-width: 360px;
  justify-content: space-between;
  border-bottom: 1px solid white;
}

.lnr-cross-circle {
  font-size: 40px;
  cursor: pointer;
  color: white;
  opacity: 0.5;
}

.lnr-download {
  display: flex;
  justify-content: center;
  padding: 10px;
  font-size: 40px;
  cursor: pointer;
  color: white;
  opacity: 0.5;
}

span:hover,
button:hover {
  transform: scale(1.1);
  opacity: 1;
  transition: 1s;
}