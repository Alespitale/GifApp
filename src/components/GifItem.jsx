import { useState } from "react";
import { GifModal } from "./GifModal";
import PropTypes from "prop-types";

export const GifItem = ({ title, url }) => {
  const [show, setShow] = useState(false);
  const handleClose = () => setShow(false);
  const handleShow = () => setShow(true);

  return (
    <div className="card-item">
      <img loading="lazy" src={url} alt={title} onClick={handleShow} />
      <GifModal url={url} title={title} show={show} onHide={handleClose}/>
    </div>
  );
};

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
