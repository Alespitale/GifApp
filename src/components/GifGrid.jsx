import { useFetchGif } from "../hooks/useFetchGif";
import { GifItem } from "./GifItem";
import { ButtonGeneral } from "./ButtonGeneral";
import PropTypes from 'prop-types'

export const GifGrid = ({ category, clear }) => {
  const { img, loading } = useFetchGif(category);

  return (
    <>
      <div className="category-container">
        <h3>{category}</h3>
        <ButtonGeneral onClick={() => clear(category)}>DELETE</ButtonGeneral>
      </div>
      <div className="card-grid">
        {loading ? (
          <h2>Loading...</h2>
        ) : (
          img.map((imagen) => <GifItem key={imagen.id} {...imagen} />)
        )}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}