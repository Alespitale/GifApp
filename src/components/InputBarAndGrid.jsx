import { AddCategory } from "./AddCategory";
import { ButtonGeneral } from "./ButtonGeneral";
import { GifGrid } from "./GifGrid";
import { useHandleCategories } from "../hooks/useHandleCategories";
import "../styles/InputBarAndGrid.css";

export const InputBarAndGrid = () => {
  const { categories, handleAddCategory, handleResetList, handleDeleteGifs } =
    useHandleCategories();

  return (
    <>
      <div className="input-bar-container">
        <div className="input-bar">
          <AddCategory
            addCategory={handleAddCategory}
            categoryName={categories}
          />
        </div>
        <ButtonGeneral data-testid="reset button" onClick={handleResetList}>
          RESET
        </ButtonGeneral>
      </div>
      {categories.map((category) => (
        <GifGrid key={category} category={category} clear={handleDeleteGifs} />
      ))}
    </>
  );
};
