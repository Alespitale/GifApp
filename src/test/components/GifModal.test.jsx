import { fireEvent, render, screen } from "@testing-library/react";
import { GifModal } from "../../components/GifModal";
import "@testing-library/jest-dom";

describe("Test of GifModal", () => {
  test("Should render the modal", () => {
    const title = "Dragon Ball";
    const url = "https://dragonball.com/goku.jpg";
    const show = true;
    const onHide = jest.fn();
    const {baseElement} = render(
      <GifModal title={title} url={url} show={show} onHide={onHide} />
    );
    expect(baseElement).toMatchSnapshot()
  });

  test("Should show title, buttons and image", () => {
    const title = "Dragon Ball";
    const url = "https://dragonball.com/goku.jpg";
    const show = true;
    const onHide = jest.fn();
    render(<GifModal title={title} url={url} show={show} onHide={onHide} />);
    expect(screen.getByText(`${title}`).innerHTML).toBe(`${title}`);
    expect(screen.getByTestId("close button")).toBeVisible();
    expect(screen.getByTestId("download button")).toBeVisible();
    const img = screen.getByRole('img')
    fireEvent.load(img)
    expect(img).toBeVisible()
  });
});
