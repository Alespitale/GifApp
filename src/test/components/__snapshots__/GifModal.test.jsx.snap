// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of GifModal Should render the modal 1`] = `
<body
  class="modal-open"
  data-rr-ui-modal-open=""
  style="padding-right: 1024px;"
>
  <div />
  <div
    class="fade modal-backdrop show"
  />
  <div
    aria-modal="true"
    class="fade modal-content modal show"
    role="dialog"
    style="display: block; padding-right: 0px;"
    tabindex="-1"
  >
    <div
      class="modal-dialog"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header modal-header"
        >
          <div
            class="modal-title modal-title h4"
          >
            Dragon Ball
          </div>
          <button
            class="close-btn"
            data-testid="close button"
          >
            <svg
              class="bi bi-x"
              fill="white"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4.646 4.646a.5.5 0 0 1 .708 0L8 7.293l2.646-2.647a.5.5 0 0 1 .708.708L8.707 8l2.647 2.646a.5.5 0 0 1-.708.708L8 8.707l-2.646 2.647a.5.5 0 0 1-.708-.708L7.293 8 4.646 5.354a.5.5 0 0 1 0-.708z"
              />
            </svg>
          </button>
        </div>
        <div
          class="modal-body"
        >
          <div>
            <img
              alt="Dragon Ball"
              src="https://dragonball.com/goku.jpg"
            />
          </div>
          <button
            class="download-btn"
            data-testid="download button"
          >
            <svg
              class="bi bi-download"
              fill="white"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M.5 9.9a.5.5 0 0 1 .5.5v2.5a1 1 0 0 0 1 1h12a1 1 0 0 0 1-1v-2.5a.5.5 0 0 1 1 0v2.5a2 2 0 0 1-2 2H2a2 2 0 0 1-2-2v-2.5a.5.5 0 0 1 .5-.5z"
              />
              <path
                d="M7.646 11.854a.5.5 0 0 0 .708 0l3-3a.5.5 0 0 0-.708-.708L8.5 10.293V1.5a.5.5 0 0 0-1 0v8.793L5.354 8.146a.5.5 0 1 0-.708.708l3 3z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
